timemachine/cpp/src/kernels/k_nonbonded.cuh

```diff
@@ -901,6 +901,19 @@ void __global__ k_nonbonded_exclusions(
 
     unsigned long long energy = 0;
 
+    int is_vanilla = (
+        lambda_offset_i == 0 &&
+        lambda_plane_i == 0 &&
+        dq_dl_i == 0 &&
+        dsig_dl_i == 0 &&
+        deps_dl_i == 0 &&
+        lambda_offset_j == 0 &&
```

Comment:
You mentioned offline that when the 4D coords are the same (which I believe the lambda values influence) we get the singularity. Do we need to check equality of the lambda values rather than just checking both are zero?

Reply:
The 4D coordinates are still computed on the fly in master, once #403 is merged this check will change.